zerver/webhooks/beeminder/tests.py

```diff
@@ -2,42 +2,39 @@
 
 from typing import Text
+from unittest.mock import MagicMock, patch
 from zerver.lib.test_classes import WebhookTestCase
 
 class BeeminderHookTests(WebhookTestCase):
     STREAM_NAME = 'beeminder'
     URL_TEMPLATE = u"/api/v1/external/beeminder?api_key={api_key}&email=AARON@zulip.com"
 
     def test_beeminder_derail(self) -> None:
-        # Important: These checks need to be in the test to avoid a race.
-        # TODO: We should mock time.time for these tests
-        current_time = float(time.time())
-        time_delta_in_hours = (float(1517759100) - current_time)/float(3600)
-
-        expected_subject = u"beekeeper"
-        expected_message = u"Hello **aaron**! I am the Beeminder bot! :octopus: \n You are going to derail \
-        from goal **gainweight** in **{:0.1f} hours** \n You need **+2 in 7 days (60)** to avoid derailing \n * Pledge: **0$** :relieved:".format(time_delta_in_hours)
-        self.send_and_test_stream_message('derail',
-                                          expected_subject,
-                                          expected_message,
-                                          content_type="application/x-www-form-urlencoded")
+        with patch('zerver.webhooks.beeminder.view.get_time') as get_time:
+            mock = MagicMock(return_value=5)
+            get_time.return_value = mock.return_value
+            expected_subject = u"beekeeper"
+            expected_message = u"Hello **aaron**! I am the Beeminder bot! :octopus: \n You are going to derail \
+from goal **gainweight** in **{:0.1f} hours** \n You need **+2 in 7 days (60)** to avoid derailing \n * Pledge: **0$** :relieved:".format(get_time())
+            self.send_and_test_stream_message('derail',
+                                              expected_subject,
+                                              expected_message,
+                                              content_type="application/x-www-form-urlencoded")
 
     def test_beeminder_derail_pm(self) -> None:
-        # Important: These checks need to be in the test to avoid a race.
-        # TODO: We should mock time.time for these tests
-        current_time = float(time.time())
-        time_delta_in_hours = (float(1517759100) - current_time)/float(3600)
-
-        self.url = self.build_webhook_url(
-            email="AARON@zulip.com",
-            username="aaron",
-            user_ip="127.0.0.1"
-        )
-        expected_message = u"I am the Beeminder bot! :octopus: \n You are going to derail from \
-        goal **gainweight** in **{:0.1f} hours** \n You need **+2 in 7 days (60)** to avoid derailing \n * Pledge: **5$**:worried:".format(time_delta_in_hours)
-        self.send_and_test_private_message('derail_pm',
-                                           expected_message,
-                                           content_type="application/json")
+        with patch('zerver.webhooks.beeminder.view.get_time') as get_time:
+            mock = MagicMock(return_value=5)
```

Comment:
Do we actually need `MagicMock` here when `get_time` is already a mock?

Reply:
So, should this be `Mock`?

Reply:
I'm wondering if you can't just say `get_time.return_value = 5`.